merlin/models/tf/blocks/retrieval/base.py

```diff
@@ -351,19 +351,27 @@ def call_outputs(
             predictions = tf.cast(predictions, tf.float32)
 
         assert isinstance(predictions, tf.Tensor), "Predictions must be a tensor"
-
-        # Positives in the first column and negatives in the subsequent columns
-        targets = tf.concat(
-            [
-                tf.ones([tf.shape(predictions)[0], 1], dtype=predictions.dtype),
-                tf.zeros(
-                    [tf.shape(predictions)[0], tf.shape(predictions)[1] - 1],
-                    dtype=predictions.dtype,
-                ),
-            ],
-            axis=1,
-        )
-        return PredictionOutput(predictions, targets, positive_item_ids=positive_item_ids)
+        # prepare targets for computing the loss and metrics
+        if isinstance(targets, tf.Tensor):
```

Comment:
As we chatted, I think it would be good to maybe extract some logic from this methid, that is becoming crowded and create some methods with good names that make it easier to understand what varies between a Two-Tower model and a Sampled softmax model